core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala

```diff
@@ -106,15 +86,16 @@ class SubmissionMonitorActor(val workspaceName: WorkspaceName,
       handleStatusResponses(response) pipeTo self
     case StatusCheckComplete(terminateActor) =>
       logger.debug(s"done checking status for submission $submissionId, terminateActor = $terminateActor")
-      if (terminateActor) stop(self)
+      // Before terminating this actor, run one more CheckCurrentWorkflowStatusCounts pass to ensure
+      // we have accurate metrics at the time of actor termination.
+      if (terminateActor) {
+        self ! CheckCurrentWorkflowStatusCounts
+        stop(self)
+      }
       else scheduleNextMonitorPass
     case CheckCurrentWorkflowStatusCounts =>
       logger.debug(s"check current workflow status counts for submission $submissionId")
-      checkCurrentWorkflowStatusCounts pipeTo self
-    case SaveCurrentWorkflowStatusCounts(statusCounts) =>
-      logger.debug(s"saving current workflow status counts for submission $submissionId")
-      currentWorkflowStatusCounts = statusCounts
-      scheduleNextCheckCurrentWorkflowStatus
+      checkCurrentWorkflowStatusCounts pipeTo parent
```

Comment:
shouldn't this go to the supervisor too, then? otherwise you're still not gonna check workflow statuses once the submission is done